app/repositories/onlinetesting/Phase1EvaluationRepository.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories.onlinetesting
+
+import model.OnlineTestCommands.Phase1TestProfile
+import model.{ ApplicationStatus, ApplicationStatuses, ProgressStatuses, SelectedSchemes }
+import model.persisted.ApplicationToPhase1Evaluation
+import reactivemongo.api.DB
+import reactivemongo.bson.{ BSONArray, BSONBoolean, BSONDocument, BSONObjectID }
+import repositories.RandomSelection
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+trait Phase1EvaluationRepository {
+  def nextApplicationReadyForPhase1ResultEvaluation: Future[Option[ApplicationToPhase1Evaluation]]
+}
+
+class Phase1EvaluationMongoRepository()(implicit mongo: () => DB)
+  extends ReactiveRepository[ApplicationToPhase1Evaluation, BSONObjectID]("application", mongo,
+    ApplicationToPhase1Evaluation.applicationToPhase1EvaluationFormats,
+    ReactiveMongoFormats.objectIdFormats) with Phase1EvaluationRepository with RandomSelection {
+
+  def nextApplicationReadyForPhase1ResultEvaluation: Future[Option[ApplicationToPhase1Evaluation]] = {
+    // TODO: Add BSONDocument("passmarkEvaluation.passmarkVersion" -> BSONDocument("$exists" -> false))
+    val query = BSONDocument("$or" -> BSONArray(
+      BSONDocument("$and" -> BSONArray(
+        BSONDocument("applicationStatus" -> ApplicationStatus.PHASE1_TESTS),
+        BSONDocument(s"progress-status.${ProgressStatuses.PHASE1_TESTS_RESULTS_RECEIVED}" -> true)
+      )),
+      BSONDocument("$and" -> BSONArray(
+        BSONDocument("applicationStatus" -> ApplicationStatus.PHASE2_TESTS),
```

Comment:
It's called "Business Logic" :)

The case is that they can be promoted to Phase2, but they may have some ambers (not evaluated schemes). In Phase2 evaluation not only check phase2 results, but also what was the result per scheme in Phase1